### PR TITLE
Improve help and mkrc

### LINKED ISSUE #206: How to update existing folder
Hi guys!

I have a question about how to renew my files in .dotfiles dir.

Example: I have an Atom editor and have already created linked folder with 'mkrc .atom' and used it on another computer. Then later I added some packages to editor and want to renew my .dotfiles/atom folder. What should I do? 'rcup atom' seems not to add new files. If I try to run 'mkrc .atom' it creates folder .dotfiles/atom/.atom for some reason.

What I am doing wrong here and what is the best practice to deal with such cases.

And still I'm here could you confirm it takes a lot of time to update large amount of files like in my case. Several hours I believe. Mac 10, latest version.

Sorry for dummy questions, new to this tool. :)

### LINKED ISSUE #254: mkrc and DOTFILES_DIRS
`man mkrc` gives:

> -d DIR  install dotfiles under the specified directory. This can be specified multiple times.

Since in the code we have:

```
for DOTFILES_DIR in $DOTFILES_DIRS $DEFAULT_DOTFILES_DIR; do
  break
done
```

I think we should add that:
*  The first `-d` option in the command overrides the setting in the RCRC file
*  Only the last one is used to deliver files, because the way it is coded, each further option overrides the previous one.

### PR DESCRIPTION
**Pull request by mat-m**
* Fix #206 : messages for -C being a copy so can't update
  * Fix #254 : expect at least one existing dotfiles directory

### CODE REVIEW

**Comment by mike-burns** [comment]
Thanks for writing this up. Seeing it in code helps clarify the idea.

My concern is around aliases. If someone has e.g.

```sh
alias mkrc='mkrc -d ~/.local/dotfiles'
```

But they want to override it in a one-off, they currently can do that,

```sh
mkrc -d /tmp/experimental-dotfiles  # after expanding the alias, is mkrc -d ~/.local/dotfiles -d /tmp/experimental-dotfiles
```

That's because the last `-d` (should) win. But if the first `-d` wins, they can't do this.

(cf. `alias rm='rm -i'`.)

What do you think? Can we clarify the docs to explain that the last `-d` wins?

**Comment by mat-m** [comment]
Currently, the precedence is  `argument > variable > mkrc > default` , but overriding an alias is not allowed, since it will bring a second folder.
`mkrc -d /tmp/experimental-dotfiles /tmp/imported-dotfiles` is a clear mistake (can't create in 2 folders), but we have no way to warn the user before creating in the last one. If it is a mistake, I'd prefer to fail early.
Moreover, going with the last enforce parsing all values, which adds risk.
I would prefer to add a warning about unaliasing `\mkrc -d /tmp/experimental-dotfiles` rather than allowing an overload where we can't enforce a sane behavior (KISS).
And it may confuse users, because mkrc  may have multiple values, but only the last one is used, while other commands will concatenate multiple -d arguments.

If you're not convinced, I will patch with last-path-wins, else I will amend mkrc doc about unalias.

**Comment by mat-m** [comment]
Solved news.md conlfict.

File: bin/mkrc.in [hunk 1]
```diff
@@ -42,6 +42,17 @@ exit_if_dangerous() {
   fi
 }
 
+only_one_dir() {
+  local dotfiles="$1"
+
+  if [ ! -z "$OPT_DOTFILES_DIR" ]; then
+    $PRINT "Only one '-d' option is allowed in mkrc."
```
Comment (mike-burns) [inline bin/mkrc.in:49]:
```suggestion
    $ERROR "Only one '-d' option is allowed in mkrc."
```

**Review by mike-burns — APPROVED** [review]
Coming back to this later, I'm onboard. I'm going to change the `$PRINT` to `$ERROR` then merge it. Thank you for this!

File: bin/mkrc.in (hunk 1 above)
Comment (mat-m) [reply bin/mkrc.in]: good catch !

**Pull request merged**